mergify_engine/actions/merge/queue.py

```diff
@@ -189,114 +183,43 @@ def get_pulls(self):
     def delete(self):
         self.redis.delete(self._redis_queue_key)
 
-    def handle_first_pull_in_queue(self, ctxt):
-        old_checks = [
-            c for c in ctxt.pull_engine_check_runs if c["name"].endswith(" (merge)")
-        ]
-
-        result = helpers.merge_report(ctxt, True)
-        if result:
-            ctxt.log.info(
-                "pull request closed in the meantime",
-                result=result,
-            )
-            self.remove_pull(ctxt.pull["number"])
-        else:
-            ctxt.log.info("updating base branch of pull request")
-            config = self.get_config(ctxt.pull["number"])
-            result = helpers.update_pull_base_branch(
-                ctxt,
-                config["strict_method"],
-                config["update_bot_account"] or config["bot_account"],
-            )
-
-            if ctxt.pull["state"] == "closed":
-                ctxt.log.info(
-                    "pull request closed in the meantime",
-                    result=result,
-                )
-                self.remove_pull(ctxt.pull["number"])
-            elif result.conclusion == check_api.Conclusion.FAILURE:
-                ctxt.log.info(
-                    "base branch update failed",
-                    result=result,
-                )
-                self._move_pull_at_end(ctxt.pull["number"])
-
-        for c in old_checks:
-            check_api.set_check_run(ctxt, c["name"], result)
-
-    @classmethod
-    def process_queues(cls):
-        with utils.get_redis_for_cache() as redis:
-            LOG.info("smart strict workflow loop start")
-            for queue_name in redis.keys("strict-merge-queues~*"):
-                queue = cls.from_queue_name(redis, queue_name)
-                try:
-                    queue.process()
-                except exceptions.MergifyNotInstalled:
-                    queue.delete()
-                except Exception:
-                    queue.log.error("Fail to process merge queue", exc_info=True)
-            LOG.info("smart strict workflow loop end")
-
-    def process(self):
-        pull_numbers = self.get_pulls()
-
-        self.log.info("%d pulls queued", len(pull_numbers), queue=list(pull_numbers))
-
-        if not pull_numbers:
-            return
-
-        pull_number = pull_numbers[0]
-
-        with github.get_client(self.owner) as client:
-            ctxt = None
-            try:
-                sub = asyncio.run(
-                    subscription.Subscription.get_subscription(client.auth.owner_id)
-                )
-                data = client.item(
-                    f"/repos/{self.owner}/{self.repo}/pulls/{pull_number}"
+    @contextlib.contextmanager
+    def pull_request_auto_refresher(self, pull_number):
+        """
+        If the first pull request in queue change we refresh it, to ensure we process
+        it's rebase
```

Comment:
```suggestion
        If the first pull request in queue change we refresh it, to ensure we process
        its rebase
```
I don't really understand the sentence though :(

Maybe:
> Make sure to refresh the first pull request in the queue if it changed during the start and end of the context.